models/repo.go

```diff
@@ -733,6 +733,16 @@ func (repo *Repository) getUsersWithAccessMode(e Engine, mode AccessMode) (_ []*
 	return users, nil
 }
 
+// BlockedByDependencies finds all Dependencies an issue is blocked by
+func (repo *Repository) BlockedByDependencies(issueID int64) (_ []*IssueDependencyIssue, err error) {
+	return repo.getBlockedByDependencies(x, issueID)
+}
+
+// BlockingDependencies returns all blocking dependencies
+func (repo *Repository) BlockingDependencies(issueID int64) (_ []*IssueDependencyIssue, err error) {
```

Comment:
1. Name `BlockingDependencies` is misleading for me. They are not blocking, but they are blocked by given issue. Something like `BlockedDependencies` looks better to me.
1. This should be part of `Issue`, not `Repository`. You can use it like `issue.GetBlockedDependencies()`/`issue.BlockedDependencies()` (or `models.GetBlockedDependencies(issueID)` for use without existing issue object) and `issue.BlockedByDependencies()`

Reply:
1. You're wrong, this function takes an issue ID and returns an array with all _issues which are blocking that given issue_. `BlockedByDependencies` on the other hand also takes an issue id and returns all issues which _are blocked by that given issue_.
2. You're right, I'll move it. Again, I am still discovering the structure of Gitea and am still discovering how things are done.... (and where, more precisely)

Reply:
@Morlinest Fixed the second one. Also added a comment fpr the first one, maybe it is more clear now.